src/c++/perf_analyzer/infer_data_manager.cc

```diff
@@ -0,0 +1,683 @@
+                          << std::endl;
+                    }
+                  }))));
+    } else {
+      cudaIpcMemHandle_t cuda_handle;
+      RETURN_IF_ERROR(
+          CreateCUDAIPCHandle(&cuda_handle, reinterpret_cast<void*>(*ptr)));
+      RETURN_IF_ERROR(backend_->RegisterCudaSharedMemory(
+          shm_region_name, cuda_handle, byte_size));
+
+      // No operation required for deleting the memory
+      shared_memory_regions_.emplace(
+          std::piecewise_construct, std::forward_as_tuple(shm_region_name),
+          std::forward_as_tuple(SharedMemoryData(
+              byte_size,
+              std::unique_ptr<uint8_t, std::function<void(uint8_t*)>>(
+                  reinterpret_cast<uint8_t*>(*ptr), [](uint8_t* memory) {}))));
+    }
+#endif  // TRITON_ENABLE_GPU
+  } else {
+    return cb::Error(
+        "CreateMemoryRegion called with invalid memory region type.",
+        pa::GENERIC_ERROR);
+  }
+
+  return cb::Error::Success;
+}
+
+cb::Error
+InferDataManager::CopySharedMemory(
+    uint8_t* input_shm_ptr, std::vector<const uint8_t*>& data_ptrs,
+    std::vector<size_t>& byte_size, bool is_shape_tensor,
+    std::string& region_name)
+{
+  if (shared_memory_type_ == SharedMemoryType::SYSTEM_SHARED_MEMORY) {
+    // Populate the region with data
+    size_t count = 0;
+    size_t offset = 0;
+    size_t max_count = is_shape_tensor ? 1 : batch_size_;
+    while (count < max_count) {
+      memcpy(input_shm_ptr + offset, data_ptrs[count], byte_size[count]);
+      offset += byte_size[count];
+      count++;
+    }
+  } else {
+#ifdef TRITON_ENABLE_GPU
+    // Populate the region with data
+    size_t count = 0;
+    size_t offset = 0;
+    size_t max_count = is_shape_tensor ? 1 : batch_size_;
+    while (count < max_count) {
+      cudaError_t cuda_err = cudaMemcpy(
+          (void*)(input_shm_ptr + offset), (void*)data_ptrs[count],
+          byte_size[count], cudaMemcpyHostToDevice);
+      if (cuda_err != cudaSuccess) {
+        return cb::Error(
+            "Failed to copy data to cuda shared memory for " + region_name +
+                " : " + std::string(cudaGetErrorString(cuda_err)),
+            pa::GENERIC_ERROR);
+      }
+      offset += byte_size[count];
+      count++;
+    }
+#endif  // TRITON_ENABLE_GPU
+  }
+  return cb::Error::Success;
+}
+
+cb::Error
+InferDataManager::CreateInferInput(
+    cb::InferInput** infer_input, const cb::BackendKind kind,
+    const std::string& name, const std::vector<int64_t>& dims,
+    const std::string& datatype)
+{
+  return cb::InferInput::Create(infer_input, kind, name, dims, datatype);
+}
+
+cb::Error
+InferDataManager::PrepareInfer(InferData& ctx)
+{
+  if (shared_memory_type_ == SharedMemoryType::NO_SHARED_MEMORY) {
+    return PrepareInferNoSharedMemoryInfer(ctx);
+  } else {
+    return PrepareSharedMemoryInfer(ctx);
+  }
+}
+
+cb::Error
+InferDataManager::PrepareInferNoSharedMemoryInfer(InferData& ctx)
```

Comment:
Don't need two 'infer' in the name. Rename them to `PrepareInfer(No)SharedMemory`

Reply:
fixed